addons/arsenal/functions/fnc_attributeAddCompatible.sqf

```diff
@@ -0,0 +1,75 @@
+#include "script_component.hpp"
+#include "..\defines.hpp"
+/*
+ * Author: mharis001
+ * Adds compatible attachments or magazines for all weapons in 3DEN attribute.
+ *
+ * Arguments:
+ * 0: Attribute controls group <CONTROL>
+ *
+ * Return Value:
+ * None
+ *
+ * Example:
+ * [CONTROL] call ace_arsenal_fnc_attributeAddCompatible
+ *
+ * Public: No
+ */
+
+params ["_controlsGroup"];
+
+private _category = lbCurSel (_controlsGroup controlsGroupCtrl IDC_ATTRIBUTE_CATEGORY);
+
+// Exit if selected category is not attachments or magazines
+if !(_category in [4, 5, 6, 7, 8]) exitWith {};
+
+private _configItems = +(uiNamespace getVariable [QGVAR(configItems), []]);
+private _attributeValue = uiNamespace getVariable [QGVAR(attributeValue), [[], 0]];
+_attributeValue params ["_attributeItems"];
+
+// Get list of all weapons in attribute items
+(_configItems select 0) params ["_primaryWeapons", "_secondaryWeapons", "_handgunWeapons"];
+private _attributeWeapons = _attributeItems select {_x in _primaryWeapons || {_x in _secondaryWeapons} || {_x in _handgunWeapons}};
+
+// Add compatible attachments or magazines to attribute
+private _cfgWeapons = configFile >> "CfgWeapons";
+private _itemsToAdd = [];
+
+if (_category == 8) then {
+    private _magazineGroups = +(uiNamespace getVariable QGVAR(magazineGroups));
```

Comment:
What is the copy for? you are not modifying this value.
L54 copies the elements into _itemsToAdd. So no problem there.